agent/src/agent_manager.rs

```diff
@@ -501,4 +544,36 @@ mod tests {
         to_manager.stop().await.unwrap();
         assert!(join!(handle).0.is_ok());
     }
+
+    // [utest->swdd~agent-sends-node-resource-availability-to-server~1]
+    #[tokio::test]
+    async fn utest_agent_manager_sends_available_resources() {
```

Comment:
I cannot see that this utest ensures that it would fail if the available resources are not sent. It only expects that `handle_update_workload` is called and that you get "some" server message. But that can also be the case due to other reasons.

Reply:
Updated the test to better reflect the actual functionality and to test that the server receives an AgentLoadStatus and not something else. Please check if it's fine.